frontend/lib/api/ApiError.ts

```diff
@@ -1,23 +1,44 @@
+import { TranslationPath } from "@kiesraad/i18n";
+
+import { ApiResult } from "./api.types";
 import { ApiResponseStatus } from "./ApiResponseStatus";
+import { ErrorReference } from "./gen/openapi";
 
 export class ApiError extends Error {
   constructor(
     public status: ApiResponseStatus.ClientError | ApiResponseStatus.ServerError,
     public code: number,
     public message = "Unknown error",
+    public fatal: boolean = true,
+    public reference: ErrorReference = "InternalServerError",
   ) {
     super(message);
   }
+}
+
+export class NetworkError extends Error {
+  constructor(public message: string) {
+    super(message);
+  }
+}
 
-  withContext(message: string): this {
+export class NotFoundError extends Error {
+  path: string;
+  message: TranslationPath;
+
+  constructor(message?: TranslationPath) {
+    super(message || "error.not_found");
+    this.message = message || "error.not_found";
+    this.path = window.location.pathname;
+  }
+
+  withMessage(message: TranslationPath): this {
     this.message = message;
 
     return this;
   }
 }
 
-export class NetworkError extends Error {
-  constructor(public message: string) {
-    super(message);
-  }
+export function isFatalError(error: ApiResult<unknown>): error is ApiError | NetworkError | NotFoundError {
```

Comment:
The type predicate is not correct here, causing my IDE to complain about unreachable code in `deleteDataEntry()`.

If error is an `ApiError` with `fatal=false` it will return false, but TypeScript cannot see that this scenario exists.

Reply:
Yeah, maybe I need to split ApiError and FatalApiError in two different types - what do you think?

Reply:
I think that would make things more straightforward indeed